### DISCUSSION TITLE
accidently source-controller deployment deleted.

### DISCUSSION THREAD

**Discussion by zachinshakya** [fluxcd/flux2#2863] [General] [answered]
Is there any way to get same deployment for source-controller in case of accidentally deletion.

**Comment by stefanprodan** [comment, marked as answer]
Rerun `flux bootstrap` and it will restore all things for you.

**Reply by kingdonb** [reply]
Another approach that should work is `kubectl apply -k ./clusters/my-cluster/flux-system` (assuming that flux-system namespace has not been destroyed as well, as it contains your deploy key secret.)
